### PR TITLE
Capture the flag quality of life changes

### PR DESCRIPTION
**Pull request by coiax**
- Spawn traps no longer fade when crossed
- The "reset" location of CTF flags is visible to observers
- Baricades regenerate at the start of every round, and when destroyed
  leave observer visible ghost barricades
- At the start of every round, the arena is cleared of debris
- The blue team shoot blue lasers, the red team shoot red lasers

:cl: coiax
rscadd: The CTF arena regenerates every round, and the blue team now shoot blue lasers.
/:cl:

### CODE REVIEW

File: code/modules/awaymissions/capture_the_flag.dm [hunk 1]
```diff
@@ -86,8 +95,26 @@
...
+	name = "red flag landmark"
+	icon_state = "banner-red"
+	desc = "This is where a red banner, used to play capture the flag \
+		would go."
+
+/obj/effect/ctf/flag_reset/blue
+	name = "blue flag landmark"
+	icon_state = "banner-blue"
+	desc = "This is where a blue banner, used to play capture the flag \
+		would go."
```
Comment (ChangelingRain) [inline code/modules/awaymissions/capture_the_flag.dm]: would go?

Comment (RemieRichards) [inline code/modules/awaymissions/capture_the_flag.dm]: Read it as the entire sentence that it is.

Comment (PKPenguin321) [inline code/modules/awaymissions/capture_the_flag.dm]: The line above this one needs a comma at the end

File: code/modules/awaymissions/capture_the_flag.dm [hunk 2]
```diff
@@ -214,9 +251,41 @@
...
+/obj/machinery/capture_the_flag/proc/clear_the_arena()
+	var/area/A = get_area(src)
+	for(var/atm in A)
+		if(!is_type_in_typecache(atm, ctf_object_typecache))
+			qdel(atm)
+
+/obj/machinery/capture_the_flag/proc/stop_ctf()
+	ctf_enabled = FALSE
+	var/area/A = get_area(src)
+	for(var/mob/M in mob_list)
```
Comment (RemieRichards) [inline code/modules/awaymissions/capture_the_flag.dm]: this should be an istypeless forloop.

File: code/modules/awaymissions/capture_the_flag.dm [hunk 3]
```diff
@@ -214,9 +251,41 @@
...
+		if((get_area(A) == A) && M.ckey in team_members)
```
Comment (RemieRichards) [inline code/modules/awaymissions/capture_the_flag.dm]: surround the ```in``` statement with parenthesis, it's got a weird operator precedence

File: code/modules/awaymissions/capture_the_flag.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/modules/awaymissions/capture_the_flag.dm]: aaaaa fucki g line breaks god damn you coiax

**Pull request merged**